WebContent crash or weird JS console behavior after a JS exception during page load

I found a way to reliably crash WebContent with `ConnectionFromClient(0x000000015960a0d0) (id=1) misbehaved (Requested non-existent console message index.), disconnecting.` error message or cause weird console behavior.

Steps to reproduce:
1. Create the following HTML file
```html
<html>
    <head>
        <title>WebContent Crash</title>
    </head>
    <body>
        <div>
            <select id="test-element">
                <option>A</option>
                <option>B</option>
            </select>
        </div>
        <script>
            throw 123;
        </script>
    </body>
</html>
```
2. Open the file in Ladybird
3. Open the Inspector window
4. Run `console.log(123)` in the JS Console
5. Browser will report that the tab has crashed

The above steps reliably cause a crash on macOS with the Debug build configuration and with the native chrome and crash or weird console behavior (results of evaluations not showing up, or results of a previous evaluation showing up) with the Qt chrome.


Edit: updated repro to not depend on a confounding bug in HTMLSelectElement which is fixed now

I was able to reproduce this. The Ladybird log output looks like this:
```
110070.817 WebContent(48967): Unhandled JavaScript exception: [TypeError] Cannot access property "selected" on undefined object "selector.selectedOptions[0]"
110070.817 WebContent(48967):     at file:///opt/ladybird/2TEST.html:3:25

110097.793 WebContent(48967): (js log) 123
110097.793 Ladybird(48964): Notified about console message we're already aware of
110097.794 WebContent(48967): ConnectionFromClient(0x000000015b7124e0) (id=1) misbehaved (Requested non-existent console message index.), disconnecting.
GrMtlDepthStencil: 0x600002c68000 (2 refs)
GrMtlRenderPipeline: 0x15b8b1000 (14 refs)
GrMtlRenderPipeline: 0x15b990400 (7 refs)
GrMtlDepthStencil: 0x600002c55180 (2 refs)
GrMtlDepthStencil: 0x600002c54c30 (2 refs)
GrMtlRenderPipeline: 0x15b99a600 (4 refs)
GrMtlDepthStencil: 0x600002c55400 (2 refs)
GrMtlRenderPipeline: 0x15d87aa00 (4 refs)
GrMtlSampler: 0x600003e7d360 (1 refs)
GrMtlRenderPipeline: 0x15d88c200 (18 refs)
GrMtlRenderPipeline: 0x15d00a000 (6 refs)
../src/1c27019d33-728a6df320.clean/src/gpu/ganesh/GrManagedResource.h:56: fatal error: "check(0 == fHashSet.count())"
110097.796 Ladybird(48964): WebContent process crashed!
```

--- a/Userland/Services/WebContent/PageClient.cpp
+++ b/Userland/Services/WebContent/PageClient.cpp
@@ -342,8 +342,13 @@ void PageClient::page_did_create_new_document(Web::DOM::Document& document)
 
 void PageClient::page_did_change_active_document_in_top_level_browsing_context(Web::DOM::Document& document)
 {
+    auto& realm = document.realm();
+
     VERIFY(m_console_clients.contains(document));
     m_top_level_document_console_client = *m_console_clients.get(document).value();
+
+    auto console_object = realm.intrinsics().console_object();
+    console_object->console().set_client(*m_top_level_document_console_client);
 }
 
 void PageClient::page_did_destroy_document(Web::DOM::Document& document)
@@ -666,7 +671,6 @@ void PageClient::initialize_js_console(Web::DOM::Document& document)
     auto& realm = document.realm();
     auto console_object = realm.intrinsics().console_object();
     auto console_client = heap().allocate_without_realm<WebContentConsoleClient>(console_object->console(), document.realm(), *this);
-    console_object->console().set_client(*console_client);
 
     m_console_clients.set(document, console_client);
 }

